src/biz/metrics.go

```diff
@@ -276,19 +249,62 @@ func (mc *MetricsFetcher) memory(mem raw.Memory) Memory {
 		softLimit = 0
 	}
 
-	return Memory{
-		MemLimitBytes:         memLimits,
-		CacheUsageBytes:       mem.Cache,
-		RSSUsageBytes:         mem.RSS,
-		UsageBytes:            mem.RSS,
-		UsagePercent:          usagePercent,
-		KernelUsageBytes:      mem.KernelMemoryUsage,
-		SwapUsageBytes:        swapUsage,
-		SwapOnlyUsageBytes:    swapOnlyUsage,
-		SwapLimitBytes:        swapLimit,
-		SoftLimitBytes:        softLimit,
-		SwapLimitUsagePercent: swapLimitUsagePercent,
+	swapLimit := mem.SwapLimit
+	if mem.SwapLimit > math.MaxInt64/2 {
+		swapLimit = 0
+	}
+
+	m := Memory{
+		MemLimitBytes:    memLimits,
+		CacheUsageBytes:  mem.Cache,
+		RSSUsageBytes:    mem.RSS,
+		UsageBytes:       mem.RSS,
+		UsagePercent:     usagePercent,
+		KernelUsageBytes: mem.KernelMemoryUsage,
+		SoftLimitBytes:   softLimit,
+		SwapLimitBytes:   swapLimit,
+	}
+
+	if mem.SwapUsage != nil {
+		m = populateSwapUsageMetrics(mem, m, swapLimit)
```

Comment:
nit: I would create a sub-struct of metric that contains these metrics and return that from the func rather than mutate Memory. Or perhaps just have the logic here.